lib/schema-controller.js

```diff
@@ -72,13 +72,25 @@ class SchemaController {
     return this.schemaBucket.getSchemas()
   }
 
+  wrapAsBuilder (compiler) {
+    return compiler
+  }
+
   // Schema Controller compilers holder
   setValidatorCompiler (validatorCompiler) {
+    this.compilersFactory = Object.assign(
+      {},
+      this.compilersFactory,
+      { buildValidator: this.wrapAsBuilder.bind(this, validatorCompiler) })
     this.validatorCompiler = validatorCompiler
     this.isCustomValidatorCompiler = true
   }
 
   setSerializerCompiler (serializerCompiler) {
+    this.compilersFactory = Object.assign(
+      {},
+      this.compilersFactory,
+      { buildSerializer: this.wrapAsBuilder.bind(this, serializerCompiler) })
```

Comment:
Ok that is interesting.   I thought it was the opposite of fragile.   Let me try to sell it to you:

What I have done here is installed a legacy adapter.   When you use the legacy API (setCustomValidator), I convert it as if you used the new and better API (the ValidatorBuilder) by wrapping it in an adapter.   Ok it looks gross as hell because it is just wrapping it in a function because both "APIs" are actually just a function call with no type definitions, but that is what it does.

I feel like this is the best possible way to handle "support old API".     Please consider it again?

-------------

Now, regarding the suggested code.  The reason I abandoned this approach (see 47e9c62ff7474831a18f59faae0d731260b650ad) is because `isCustomValidatorCompiler` is true **both** for the legacy case `setCustomValidator` and the modern case of customization via a custom builder in `opts`.   In the latter case, we definitely want to treat it as a normal builder and call it every time.   Telling the difference would have involved checking the builder function for identity with the default builder function which seems like a terrible idea to me.

Reply:
In my own code, I write extremely verbose comments (just because I hate coming back to it later otherwise :) ) so you may have to edit these down to something you are willing to have in your code base.   I explained in detail why the code is as it is.